massa-execution-exports/src/types.rs

```diff
@@ -19,17 +19,43 @@ pub struct ExecutionOutput {
     pub events: EventStore,
 }
 
-/// structure describing a read-only execution request
+/// structure describing different types of read-only execution request
+#[derive(Debug, Clone)]
+pub enum ReadOnlyRequest {
```

Comment:
I would suggest 

```rust
/// structure describing different types of read-only execution request
#[derive(Debug, Clone)]
pub enum ReadOnlyRequest {
/// Maximum gas to spend in the execution.
    pub max_gas: u64,
    /// The simulated price of gas for the read-only execution.
    pub simulated_gas_price: Amount,
    /// Call stack to simulate, older caller first
    pub call_stack: Vec<ExecutionStackElement>,
    /// Target of the request
    target: ReadOnlyTarget
}

enum ReadOnlyTarget {
    BytecodeExecution(Vec<u8>),
    FunctionCall {
        /// Target address
        pub target_addr: Address,
        /// Target function
        pub target_func: String,
        /// Parameter to pass to the target function
        pub parameter: String,
        }
}
```

That way you can also unify the two `execute_readonly_` functions in `execution.rs`(they would only differ at the end when calling into runtime)

Reply:
done

Reply:
very good idea by the way !